jwtmiddleware.go

```diff
@@ -222,14 +222,13 @@ func (m *JWTMiddleware) CheckJWT(w http.ResponseWriter, r *http.Request) error {
 	if !parsedToken.Valid {
 		m.logf("Token is invalid")
 		m.Options.ErrorHandler(w, r, "The token isn't valid")
-		return fmt.Errorf("Token is invalid")
+		return errors.New("Token is invalid")
 	}
 
 	m.logf("JWT: %v", parsedToken)
 
 	// If we get here, everything worked and we can set the
 	// user property in context.
-	context.Set(r, m.Options.UserProperty, parsedToken)
-
+	r.WithContext(context.WithValue(r.Context(), m.Options.UserProperty, parsedToken))
```

Comment:
r.WithContext() function returns a new request, which should be passed to next http.Handler